File: AkkaSupervisorExample/src/main/java/org/akka/essentials/supervisor/example2/WorkerActor1.java
package org.akka.essentials.supervisor.example2;

import org.akka.essentials.supervisor.example2.MyActorSystem2.Result;

import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;

public class WorkerActor1 extends UntypedActor {
	LoggingAdapter log = Logging.getLogger(getContext().system(), this);
	private int state = 0;

	@Override
	public void preStart() {
		log.info("Starting WorkerActor1 instance hashcode #" + this.hashCode());
	}

	public void onReceive(Object o) throws Exception {
		if (o == null) {
			throw new NullPointerException("Null Value Passed");
		} else if (o instanceof Integer) {
			Integer value = (Integer) o;
			if (value <= 0) {
				throw new ArithmeticException("Number equal or less than zero");
			} else
				state = value;
		} else if (o instanceof Result) {
			getSender().tell(state);
		} else {
			throw new IllegalArgumentException("Wrong Arguement");
		}
	}

	@Override
	public void postStop() {
		log.info("Stopping WorkerActor1 instance hashcode #" + this.hashCode());

	}
}

File: AkkaSupervisorExample/src/main/scala/org/akka/essentials/supervisor/example2/WorkerActor1.scala
package org.akka.essentials.supervisor.example2
import akka.actor.actorRef2Scala
import akka.actor.Actor
import akka.actor.ActorLogging

class WorkerActor1 extends Actor with ActorLogging {
  import org.akka.essentials.supervisor.example2.Result

  var state: Int = 0;

  override def preStart() {
    log.info("Starting WorkerActor instance hashcode #" + this.hashCode());
  }
  override def postStop() {
    log.info("Stopping WorkerActor instance hashcode #" + this.hashCode());
  }
  def receive: Receive = {
    case value: Int =>
      if (value <= 0)
        throw new ArithmeticException("Number equal or less than zero");
      else
        state = value;
    case result: Result =>
      sender ! state
    case ex: NullPointerException =>
      throw new NullPointerException("Null Value Passed");
    case _ =>
      throw new IllegalArgumentException("Wrong Argument");
  }
}